core/src/com/unciv/logic/city/CityStats.kt

```diff
@@ -550,33 +549,38 @@ class CityStats(val cityInfo: CityInfo) {
             entry.science *= statPercentBonusesSum.science.toPercent()
         }
 
-
         /* Okay, food calculation is complicated.
         First we see how much food we generate. Then we apply production bonuses to it.
         Up till here, business as usual.
         Then, we deduct food eaten (from the total produced).
-        Now we have the excess food, which has its own things. If we're unhappy, cut it by 1/4.
-        Some policies have bonuses for excess food only, not general food production.
-         */
+        Now we have the excess food, to which "growth" modifiers apply
+        Some policies have bonuses for growth only, not general food production. */
 
         updateFoodEaten()
         newFinalStatList["Population"]!!.food -= foodEaten
 
         var totalFood = newFinalStatList.values.map { it.food }.sum()
 
-        if (isUnhappy && totalFood > 0) { // Reduce excess food to 1/4 per the same
-            val foodReducedByUnhappiness = Stats(food = totalFood * (-3 / 4f))
-            baseStatList = LinkedHashMap(baseStatList).apply { put("Unhappiness", foodReducedByUnhappiness) } // concurrency-safe addition
-            newFinalStatList["Unhappiness"] = foodReducedByUnhappiness
-        }
-
-        totalFood = newFinalStatList.values.map { it.food }.sum() // recalculate because of previous change
-
-        // Since growth bonuses are special, (applied afterwards) they will be displayed separately in the user interface as well.
-        if (totalFood > 0 && !isUnhappy) { // Percentage Growth bonus revoked when unhappy per https://forums.civfanatics.com/resources/complete-guide-to-happiness-vanilla.25584/
-            val foodFromGrowthBonuses = getGrowthBonusFromPoliciesAndWonders() * totalFood
-            newFinalStatList.add("Growth bonus", Stats(food = foodFromGrowthBonuses))  // Why Policies? Wonders can also provide this?
-            totalFood = newFinalStatList.values.map { it.food }.sum() // recalculate again
+        // Apply growth modifier only when positive food
+        if (totalFood > 0) {
+            // Since growth bonuses are special, (applied afterwards) they will be displayed separately in the user interface as well.
+            // All bonuses except We Love The King do apply even when unhappy
+            val foodFromGrowthBonuses = Stats(food = getGrowthBonusFromPoliciesAndWonders() * totalFood)
```

Comment:
According to
https://forums.civfanatics.com/resources/complete-guide-to-happiness-vanilla.25584/
All positive growth modifiers are not applied when -10 < happiness < 0 (as was the case in the old code)
Is that source only valid for vanilla?

Reply:
Seems that it was never the case, relevant lines:
[Vanilla, starting on 5646](https://github.com/Gedemon/Civ5-DLL/blob/master/CvGameCoreDLL/CvCity.cpp)
[G&K, starting on 6294](https://github.com/Gedemon/Civ5-DLL/blob/master/CvGameCoreDLL_Expansion1/CvCity.cpp)
only specifically WLTKD bonus is supressed.